setup.py

```diff
@@ -52,7 +52,7 @@ def get_version(filename):
         'jschema_to_python~=1.2.3',
         'sarif-om~=1.0.4'
     ],
-    python_requires='>=3.6, <=4.0, !=4.0',
+    python_requires='>=3.7, <=4.0, !=4.0',
```

Comment:
I do wonder if this is a bit early with 3.6 still being ~12% of daily downloads:
https://pypistats.org/packages/cfn-lint
Is this enforced or just support guidance?

Reply:
Valid question.

Python 3.6 was EOL December 2021 - https://endoflife.date/python
SAM has dropped support along with boto3 and botocore - https://aws.amazon.com/blogs/developer/python-support-policy-updates-for-aws-sdks-and-tools/
Lambda has dropped support 

I know its a good chunk of downloads but how long can we reasonably maintain it?

Reply:
you are not testing with Python 3.6 https://github.com/aws-cloudformation/cfn-lint/blob/main/.github/workflows/test.yaml#L11

Reply:
Also installing currently on Python 3.6

```
...
Collecting aws-sam-translator>=1.50.0 (from cfn-lint)
  Could not find a version that satisfies the requirement aws-sam-translator>=1.50.0 (from cfn-lint) (from versions: 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.6.0, 1.6.1, 1.7.0, 1.8.0, 1.9.0, 1.9.1, 1.10.0, 1.11.0, 1.12.0, 1.13.0, 1.13.1, 1.13.2, 1.14.0, 1.15.0, 1.15.1, 1.16.0, 1.17.0, 1.18.0, 1.19.0, 1.19.1, 1.20.0, 1.20.1, 1.21.0, 1.22.0, 1.23.0, 1.24.0, 1.25.0, 1.26.0, 1.27.0, 1.28.1, 1.29.0, 1.30.0, 1.30.1, 1.31.0, 1.32.0, 1.33.0, 1.34.0, 1.35.0, 1.36.0, 1.37.0, 1.38.0, 1.39.0, 1.40.0, 1.42.0)
No matching distribution found for aws-sam-translator>=1.50.0 (from cfn-lint)
```

Reply:
This makes v0.58.4 the last version that will install properly.

Reply:
Looking at the numbers there are people installing versions of `cfn-lint` beyond `v0.58.4` but they must doing some work to make that happen